ops/main.py

```diff
@@ -11,529 +11,18 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+"""Support legacy ops.main.main() import."""
 
-"""Main entry point to the framework.
-
-Note that this module is callable, and calls the :func:`ops.main.main` function.
-This is so that :code:`import ops` followed by :code:`ops.main(MyCharm)` works
-as expected.
-"""
-
-import logging
-import os
-import shutil
-import subprocess
-import sys
 import warnings
-from pathlib import Path
-from typing import Any, Dict, List, Optional, Tuple, Type, Union, cast
+from typing import Optional, Type
 
 import ops.charm
-import ops.framework
-import ops.model
-import ops.storage
-from ops.charm import CharmMeta
-from ops.jujucontext import _JujuContext
-from ops.log import setup_root_logging
```

Comment:
I worry about these (other than `_JujuContext`). No-one *should* be doing `from ops.main import CharmMeta` but as we saw with the _JujuContext PR, sometimes people are importing from the wrong place, and I think we should be careful to not break those (it's really our fault for not defining `__all__` or making them private in the first place).

I think it is ok to assume no-one is doing `from ops.main import ops`. So I think we can just import and export `CharmMeta` and `setup_root_logging` with a note that it's for backwards compatibility, just like we did in the `_JujuContext` PR.

Reply:
I've decided to re-export the set of symbols that Scenario 6 uses specifically.

Rationale: many charms use Scenario and we shouldn't potentially break a whole bunch of charms if they pin dependencies asymmetrically (compatible ops, exact scenario); and to decouple Scenario 6 update from this PR in calendar time.